File: lc-dc-sep/lc-dc-sep12/solution.cpp
// Problem: 1684. Count the Number of Consistent Strings

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int count = 0;
        unordered_set<char> allowedChars(allowed.begin(), allowed.end());
        for(auto& word: words){
            bool isConsistent = true;
            for(int i=0; i<word.size(); i++){
                char currWord = word[i];
                if(allowedChars.find(currWord) == allowedChars.end()){
                    isConsistent = false;
                    break;
                }
            }
            if(isConsistent) count++;
        }
        return count;
    }
};

File: lc-dc-sep/lc-dc-sep12/solution.py
# Problem: 1684. Count the Number of Consistent Strings

class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed = set(allowed)
        count = 0
        for word in words:
            if all(char in allowed for char in word):
                count += 1

        return count